src/encoded/commands/deploy.py

```diff
@@ -23,7 +27,164 @@
         'NoDevice': "",
     },
 ]
+class spot_client(object):
+    def __init__(self):
+        self._spotClient = None
+    
+    @property
+    def spotClient(self):
+        return self._spotClient
+
+    @spotClient.setter
+    def spotClient(self, value):
+        self._spotClient = value
+
+def get_spot_id(instance, client):
+    
+    for key, value in instance.items():
+        if key == 'SpotInstanceRequests':
+            for item in value:
+                for i in item:
+                    if i == 'SpotInstanceRequestId':
+                        SpotInstanceRequestId=item[i]
+    return SpotInstanceRequestId
+
+def get_spot_code(instance, client, spot_id):
+    request = client.describe_spot_instance_requests(SpotInstanceRequestIds=[get_spot_id(instance, client)])
+    for key, value in request.items():
```

Comment:
There is a way cleaner way to do this loop.
I think what you are looking for is:
code_status = request['SpotInstnceRequests]['Status']['Code']
I don't think you need to iterate.